### ISSUE TITLE
z-order issues?

### ISSUE THREAD

**Issue by 4aiman** [groverburger/g3d#11]
I doubt I can explain better than this little test project of mine.
- Get the test project [here](https://www.dropbox.com/s/3gd5fnus75o55jx/g3d_test.zip?dl=1)
- Go inside the left-most cuboid
- See how a later-added model shows up above the 1st one.

**Comment by 4aiman** [comment]
Also, the normals in the demo are flipped.
Plus, due to the issue, the background is being drawn above everything else (even if I go outside the BG sphere)
![image](https://user-images.githubusercontent.com/2913210/105029592-97cd2c00-5a63-11eb-80f3-00d235310384.png)

**Comment by groverburger** [comment]
This is an interesting problem - your `g3d_test.zip` project seems to work fine for me. The near cubes appear to overlap far cubes as they should.

![1611180670](https://user-images.githubusercontent.com/19754251/105247289-6880f280-5b29-11eb-8b9b-de1ed49a7946.png)

This backwards depth-sorting issue appears to be a graphics card problem as it only occurs on some computers. What GPU are you running this on? A quick fix for making g3d work for you would probably be changing the depth mode from the default `love.graphics.setDepthMode("lequal", true)` to `love.graphics.setDepthMode("gequal", true)`, changing the [depth comparison](https://love2d.org/wiki/CompareMode) in the [depth mode](https://love2d.org/wiki/love.graphics.setDepthMode).

In order to work around this problem in the long run, some depth testing tricks may have to be pulled on g3d's initialization to determine which depth mode would be correct.

As for the normals in the built-in demo, the models were flipped to account for their backwards UV coordinates. I just pushed a change to correct this on the model itself instead of flipping the model at runtime, although the normals of the models shouldn't matter as they aren't used in any shader.

**Comment by 4aiman** [comment]
Running GTX1660 Super with the Studio Driver v 460.89.
The same issue was present on an older game-ready driver (450-ish).

The ```love.graphics.setDepthMode("gequal", true)``` did nothing, the picture is the same in any of the two modes.
Tried both mine and demo projects.

Also, not that it matters now (as I can see the difference on your screenshot), the idea behind the test project is to be inside the "cubes".

I think I've found the source of the issue.

```print( love.graphics.getRendererInfo( ))``` yields ```OpenGL  3.3.0 NVIDIA 460.89     NVIDIA Corporation      GeForce GTX 1660 SUPER/PCIe/SSE2``` which is weird, since the GPU is capable of OGL 4.6.

Love2d v 0.10.2 returns this ```OpenGL  4.6.0 NVIDIA 460.89     NVIDIA Corporation      GeForce GTX 1660 SUPER/PCIe/SSE2``` which is perfectly fine.

And when I try [Angle](https://love2d.org/wiki/GLES_Shader_Testing_in_Desktop) with 11.3, I'm getting this ```OpenGL ES       OpenGL ES 3.0.0 (ANGLE 2.1.14699 git hash: 60015ff672d9)        Google Inc.     ANGLE (NVIDIA GeForce GTX 1660 SUPER Direct3D11 vs_5_0 ps_5_0)``` and the issue persists.

I've tried x86 and x64 versions of love2d, to no avail.
Since love2d v 0.10.2 gets it right, it's probably something introduced in love 11. And it goes back to 11.0.

Guess, this can be closed then. Sorry for any inconvenience.
I'll re-post this on love2d forums.

Thanks! 🥇
